docs/tutorials/03_shaders/viz_sdf_cylinder.py

```diff
@@ -0,0 +1,347 @@
+# In order to see how cylinders are made, we set different resolutions (number
+# of sides used to define the bases of the cylinder) to see how it changes the
+# surface of the primitive.
+
+radius = .5
+height = 1
+
+cylinders_8 = actor.cylinder(centers[:3], dirs[:3], colors[:3], radius=radius,
+                             heights=height, capped=True, resolution=8)
+cylinders_16 = actor.cylinder(centers[3: 6], dirs[3: 6], colors[3: 6],
+                              radius=radius, heights=height, capped=True,
+                              resolution=16)
+cylinders_32 = actor.cylinder(centers[6: 9], dirs[6: 9], colors[6: 9],
+                              radius=radius, heights=height, capped=True,
+                              resolution=32)
+
+###############################################################################
+# Next, we set up a new scene to add and visualize the actors created.
+
+scene = window.Scene()
+
+scene.add(cylinders_8)
+scene.add(cylinders_16)
+scene.add(cylinders_32)
+
+interactive = False
+
+if interactive:
+    window.show(scene)
+
+window.record(scene, size=(600, 600), out_path='viz_poly_cylinder.png')
+
+###############################################################################
+# Visualize the surface geometry representation for the object.
+
+cylinders_8.GetProperty().SetRepresentationToWireframe()
+cylinders_16.GetProperty().SetRepresentationToWireframe()
+cylinders_32.GetProperty().SetRepresentationToWireframe()
+
+if interactive:
+    window.show(scene)
+
+###############################################################################
+# Then we clean the scene to render the boxes.
+
+window.record(scene, size=(600, 600), out_path='viz_poly_cylinder_geom.png')
+
+scene.clear()
+
+###############################################################################
+# Cylinder using SDF
+# ================
+# We will use the ray marching algorithm to render the SDF primitive using
+# shaders. Signed Distance Functions (SDFs) are mathematical functions that
+# determine the distance from a point in space to a surface. Ray marching is
+# a technique where you step along a ray in order to find intersections with
+# solid geometry. Objects in the scene are defined by SDF, and because we
+# don’t use polygonal meshes it is possible to define perfectly smooth
+# surfaces and allows a faster rendering in comparison to polygon-based
+# modeling.
+
+###############################################################################
+# Now we create cylinders using box actor and SDF implementation on shaders.
+# For this, we first create a box actor.
+
+box_actor = actor.box(centers=centers, directions=dirs, colors=colors,
+                      scales=(height, radius*2, radius*2))
+
+box_actor.GetProperty().SetRepresentationToWireframe()
+scene.add(box_actor)
+
+if interactive:
+    window.show(scene)
+
+window.record(scene, size=(600, 600), out_path='viz_sdf_cylinder_box.png')
+
+scene.clear()
```

Comment:
Given the description this could be removed so you can focus on describing the ray marching implementation.